### PR TITLE
Updated the github workflow to generate the JSON schemas and Definitions for Kuma components

### PR DESCRIPTION
**Pull request by Shreyas220**
Signed-off-by: Shreyas220 <shreyas.ny@gmail.com>

**Description**
Updated the github workflow to generate the JSON schemas and Definitions for Kuma components
 #174  
**Notes for Reviewers**
i no longer have access to #180, sorry for the inconvenience 

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: .github/workflows/update-oam-defs.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        with:
+          repository: kumahq/kuma 
+          excludes: prerelease, draft
+      - name: Check out code
+        uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+      - name: Get CURRENT_KUMA_VERSION from .env
+        run: |
+          cat .github/.env > $GITHUB_ENV
```
Comment (sayantan1413) [inline .github/workflows/update-oam-defs.yml:26]: Hey @Shreyas220 consider changing this to https://github.com/meshery/meshery-osm/blob/master/.github/workflows/update-oam-defs.yml#L24-L31

File: .github/workflows/update-oam-defs.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            | ."spec"."metadata"."k8sAPIVersion" = "'$(printf $apiGroup/$apiVersion $apiGroup $apiVersion)'"
+            | ."spec"."metadata"."k8sKind" = "'$nameUpper'"
+            | ."spec"."definitionRef"."name"="'$definitionRef'"' > $dirPath/$nameLower.kuma_definition.json
+          done
+      - name: Cleanup
+        run: |
+          rm -rf util kuma-*
+      - name: Update KUMA version in .env
+        run: |
+          echo "CURRENT_KUMA_VERSION=$NEW_KUMA_VERSION" > .github/.env
```
Comment (sayantan1413) [inline .github/workflows/update-oam-defs.yml:84]: We can delete this as we don't need to take the version from .env that was present in .github.

File: .github/workflows/update-oam-defs.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            definitionRef=$(printf %s.kuma.meshery.layer5.io $name)
+            ./kubeopenapi-jsonschema --location ~/kuma.yaml -t yaml --filter '$[?(@.kind=="CustomResourceDefinition" && @.spec.names.kind=='$t')]..openAPIV3Schema' --o-filter "$[]" -o json | jq '.[] | .version = "'$NEW_KUMA_VERSION'" | .title = "'$(printf %skuma $nameUpper)'"' > ../templates/oam/workloads-$NEW_KUMA_VERSION/$(printf %s.kuma.meshery.layer5io.schema.json $nameLower)
+            echo $template | jq '  ."metadata"."name" = "'$(printf %s.Kuma $nameUpper)'"
+            | ."spec"."metadata"."k8sAPIVersion" = "'$(printf $apiGroup/$apiVersion $apiGroup $apiVersion)'"
+            | ."spec"."metadata"."k8sKind" = "'$nameUpper'"
+            | ."spec"."definitionRef"."name"="'$definitionRef'"' > $dirPath/$nameLower.kuma_definition.json
+          done
+      - name: Cleanup
+        run: |
+          rm -rf util kuma-*
```
Comment (sayantan1413) [inline .github/workflows/update-oam-defs.yml:81]: Consider changing it to `rm -rf util`

**Review by sayantan1413 — CHANGES_REQUESTED** [review]
Great work @Shreyas220🎉 . Some minor changes requested 😃

File: .github/workflows/update-oam-defs.yml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+          nvm install v14.15.3
+          nvm install-latest-npm
+          git clone https://github.com/layer5io/kubeopenapi-jsonschema util
+          cd util
+          npm i; npm i nexe -g
+          make linux
+      - name: Generate Workload definitions
+        run: |
+          mkdir -p ./templates/oam/workloads-$NEW_KUMA_VERSION
+          cd util
```
Comment (sayantan1413) [inline .github/workflows/update-oam-defs.yml:62]: Consider changing this in this way https://github.com/meshery/meshery-osm/blob/master/.github/workflows/update-oam-defs.yml#L66-L70

**Review by sayantan1413 — CHANGES_REQUESTED** [review]

File: .github/workflows/update-oam-defs.yml [hunk 5]
```diff
@@ -58,8 +63,11 @@ jobs:
           make linux
       - name: Generate Workload definitions
         run: |
-          mkdir -p ./templates/oam/workloads-$NEW_KUMA_VERSION
+          dirPath=../templates/oam/workloads/workloads-$NEW_OSM_VERSION
           cd util
+          mkdir -p $dirPath
+          meshName=osm
```
Comment (sayantan1413) [inline .github/workflows/update-oam-defs.yml:69]: meshName=kuma

File: .github/workflows/update-oam-defs.yml [hunk 6]
```diff
@@ -58,8 +63,11 @@ jobs:
           make linux
       - name: Generate Workload definitions
         run: |
-          mkdir -p ./templates/oam/workloads-$NEW_KUMA_VERSION
+          dirPath=../templates/oam/workloads/workloads-$NEW_OSM_VERSION
```
Comment (sayantan1413) [inline .github/workflows/update-oam-defs.yml:66]: $NEW_KUMA_VERSION

**Review by sayantan1413 — CHANGES_REQUESTED** [review]

**Comment by Shreyas220** [comment]
@sayantan1413 @DelusionalOptimist The components aren't being generated, it is getting skipped

**Comment by sayantan1413** [comment]
Hey @Shreyas220 can you please replicate this changes https://github.com/sayantan1413/meshery-kuma/blob/master/.github/workflows/update-oam-defs.yml to your current change. I think this will solve the problem of the workflow getting skipped.

**Comment by Shreyas220** [comment]
https://github.com/Shreyas220/meshery-kuma/tree/master/templates/oam/workloads/workloads-1.2.2

**Review by sayantan1413 — CHANGES_REQUESTED** [review]

**Comment by leecalcote** [comment]
@Shreyas220 will you resolve any comments that you have addressed?

**Comment by DelusionalOptimist** [comment]
@sayantan1413 looking good? anything to add?

**Review by DelusionalOptimist — APPROVED** [review]
LGTM :rocket:.
Thanks @Shreyas220 and @sayantan1413 :raised_hands:

**Pull request merged**